components/ParsePOSPayment/Receive-Invoice.tsx

```diff
@@ -0,0 +1,159 @@
+import copy from "copy-to-clipboard"
+import { useRouter } from "next/router"
+import React from "react"
+import Image from "react-bootstrap/Image"
+import OverlayTrigger from "react-bootstrap/OverlayTrigger"
+import Tooltip from "react-bootstrap/Tooltip"
+import { QRCode } from "react-qrcode-logo"
+import { useTimer } from "react-timer-hook"
+
+import useCreateInvoice from "../../hooks/use-Create-Invoice"
+import { LnInvoiceObject } from "../../lib/graphql/index.types.d"
+import useSatPrice from "../../lib/use-sat-price"
+import { ACTION_TYPE } from "../../pages/merchant/_reducer"
+import { formatOperand } from "../../utils/utils"
+import PaymentOutcome from "../PaymentOutcome"
+import styles from "./parse-payment.module.css"
+
+interface Props {
+  recipientWalletCurrency?: string
+  walletId: string | undefined
+  state: React.ComponentState
+  dispatch: React.Dispatch<ACTION_TYPE>
+}
+
+function ReceiveInvoice({ recipientWalletCurrency, walletId, dispatch }: Props) {
+  const { usdToSats } = useSatPrice()
+  const { amount, currency } = useRouter().query
+  const [copied, setCopied] = React.useState<boolean>(false)
+
+  const { createInvoice, data, error, loading } = useCreateInvoice({
+    recipientWalletCurrency,
+  })
+  const paymentAmount = React.useMemo(() => {
+    if (currency === "USD") {
+      return amount
+    }
+    return formatOperand(usdToSats(Number(amount)).toFixed().toString())
+  }, [amount, currency, usdToSats])
+
+  const generateInvoice = React.useCallback(() => {
```

Comment:
I think that ```useCallback``` is unnecessary here. You could take the dependency array of ```[paymentAmount, walletId, createInvoice]``` and move it to line 48 and just generate the invoice in the ```useEffect``` of line 46. It seems you are using two hooks to accomplish what could be done with just one hook.

Reply:
Right. changed that.